src/util/painterscope.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Ferran Pujol Camins on 2019-06-22.
+//
+
+#pragma once
+
+#include <QPainter>
+
+// This class provides RAII style management of a QPainter properties.
+//
+// PainterScope will save the painter state on creation, and restore it
+// on destruction.
+class PainterScope final {
+  public:
+    PainterScope() = delete;
+    PainterScope(const PainterScope&) = delete;
+    PainterScope& operator=(const PainterScope&) = delete;
+
+    explicit PainterScope(QPainter* pPainter)
+            : m_pPainter(pPainter) {
+        if (m_pPainter) {
+            m_pPainter->save();
+        }
+    }
+
+    ~PainterScope() {
+        if (m_pPainter) {
+            m_pPainter->restore();
+        }
+    }
+
+  private:
+    QPainter* m_pPainter;
```

Comment:
Since the pointer is never never changed nor reset you can make it immutable by adding `const`. This implicitly disables operator=.

Reply:
I prefer to keep the explicitly deleted constructor though.